var/spack/repos/builtin/packages/whip/package.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack.package import *
+
+
+class Whip(CMakePackage, CudaPackage, ROCmPackage):
+    """whip is a small C++ abstraction layer for CUDA and HIP."""
+
+    homepage = "https://github.com/eth-cscs/whip/"
+    url = "https://github.com/eth-cscs/whip/archive/0.0.0.tar.gz"
+    git = "https://github.com/eth-cscs/whip.git"
+    maintainers = ["msimberg", "rasolca"]
+
+    version("main", branch="main")
+
+    depends_on("cmake@3.22:", type="build")
+
+    # Exactly one of +cuda and +rocm need to be set
+    conflicts("~cuda ~rocm")
+    conflicts("+cuda +rocm")
+
+    def cmake_args(self):
+        if self.spec.satisfies("+cuda"):
+            return [self.define("WHIP_BACKEND", "CUDA")]
+        elif self.spec.satisfies("+rocm"):
+            return [self.define("WHIP_BACKEND", "HIP")]
+        else:
+            raise InstallError("whip requires exactly one of +cuda and +rocm to be set")
```

Comment:
This cannot happen due to the conflicts above

Reply:
I'm aware. However, I left it in there to make sure the conflicts aren't modified wrongly in the future. Perhaps too conservative? Happy to remove it though if you think it's unnecessary.

Reply:
Yeah, let's just remove it. Then I'm happy to merge this

Reply:
Removed and pushed.